src/model.ts

```diff
@@ -587,7 +683,7 @@ export class SpecQueryModel {
       if (isEnumSpec(encQ.channel)) return null;
 
       // assemble other property into a field def.
-      const PROPERTIES = [Property.AGGREGATE, Property.AXIS, Property.BIN, Property.SORT, Property.SCALE, Property.TIMEUNIT, Property.FIELD, Property.TYPE];
+      const PROPERTIES = [Property.AGGREGATE, Property.AXIS, Property.BIN, Property.LEGEND, Property.SORT, Property.SCALE, Property.TIMEUNIT, Property.FIELD, Property.TYPE];
```

Comment:
Please use this order 
AGGREGATE, BIN, TIMEUNIT,
FIELD, TYPE,
SCALE, SORT,
AXIS, LEGEND